### PR TITLE
chore: migration to Mapbox for event location

### LINKED ISSUE #1969: Replace google places API with Mapbox API
**Describe the feature you'd like**
Documentation: https://docs.mapbox.com/android/plugins/overview/places/

**Would you like to work on the issue?**

Open for all

### PR DESCRIPTION
**Pull request by rishabh-997**
Fixes #1969 

Checklist:

- [x] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

gif for the change:

gif

### CODE REVIEW

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-app/open-event-orga-app/pullRequest?prid=4593769&bid=15423010)

File: app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 1]
```diff
@@ -32,6 +31,7 @@
 import android.text.Editable;
 import android.text.TextUtils;
 import android.text.TextWatcher;
+import android.util.Log;
```
Comment (iamareebjamal) [inline app/src/playStore/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Codacy Issue found: [Avoid unused imports such as 'android.util.Log'](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4593769)

**Comment by iamareebjamal** [comment]
Read my comment from previous PR

**Comment by rishabh-997** [comment]
> Read my comment from previous PR

are you asking me to merge both the variants into one single file ???

**Comment by rishabh-997** [comment]
> Read my comment from previous PR

sir, in the F-Droid variant, there is an option for **Payment Preference** which ain't in play store variant.
This is the difference in event wizard... So how shall I proceed ???

**Comment by iamareebjamal** [comment]
How much difference is there

**Comment by rishabh-997** [comment]
> How much difference is there



> How much difference is there

There is onClickListener in fdroid variant which triggers a Payment preference fragment to be displayed on the screen. Around 10-15 lines of code

**Comment by iamareebjamal** [comment]
Merge it and conditionally add the listener

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-app/open-event-orga-app/pullRequest?prid=4593769&bid=15423010)

File: app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 2]
```diff
@@ -113,6 +114,12 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 
         setHasOptionsMenu(true);
 
+        if(BuildConfig.FLAVOR.equals("fdroid")) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4593769)

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-app/open-event-orga-app/pullRequest?prid=4593769&bid=15423010)

File: app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 3]
```diff
@@ -113,6 +114,12 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 
         setHasOptionsMenu(true);
 
+        if("fdroid".equals(BuildConfig.FLAVOR)) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4593769)

**Comment by rishabh-997** [comment]
> Merge it and conditionally add the listener

Both the variants have been merged, you can review the PR sir

File: app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java (hunk 3 above)
Comment (iamareebjamal) [reply app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]:
```suggestion
        if ("fdroid".equals(BuildConfig.FLAVOR)) {
```

**Comment by iamareebjamal** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/open-event-orga-app/open-event-orga-app/pullRequest?prid=4593769&bid=15423010)

File: app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 4]
```diff
@@ -113,6 +114,12 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
 
         setHasOptionsMenu(true);
 
+        if ("fdroid".equals(BuildConfig.FLAVOR)) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java:117]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4593769)

File: app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 5]
```diff
@@ -317,7 +324,7 @@ private void setupPlacesAutocomplete() {
 
         binding.form.selectLocationButton.setOnClickListener(view -> {
 
-            if (mapboxAccessToken.equals("YOUR_ACCESS_TOKEN")) {
+            if ("YOUR_ACCESS_TOKEN".equals(mapboxAccessToken)) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java:327]: Codacy Issue found: [Position literals first in String comparisons](https://app.codacy.com/app/fossasia/open-event-orga-app/pullRequest?prid=4593769)

**Comment by iamareebjamal** [comment]
Test that both variants are working as expected

**Comment by rishabh-997** [comment]
> Test that both variants are working as expected

tested both variants for **location picker** as well as **payment references**

**Review by iamareebjamal — APPROVED** [review]

**Pull request merged**